### ISSUE TITLE
Move the contract expiry fixup to the decoder

### ISSUE THREAD

**Issue by dmoklaf** [erdewit/ib_insync#650]
This is a follow-up issue after #649

To allow API clients to call directly the underlying IB reqContractDetails API, the piece of code related to fixing up contracts' expiry dates needs to be moved into a transversal place, the decoder:
https://github.com/erdewit/ib_insync/blob/7337c9a3dd93b3b0eb9b129f77956401aad70a05/ib_insync/ib.py#L1860

**Comment by erdewit** [comment]
This is already done some time ago!

**Comment by dmoklaf** [comment]
Ah sorry for the noise, I didn't check master
